airflow/compat/hashlib_wrapper.py

```diff
@@ -0,0 +1,45 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import hashlib
+import sys
+from typing import Any
+
+# Check if "usedforsecurity" is available for hashlib
+if sys.version_info >= (3, 9):
+    HAS_USEDFORSECURITY = True
+else:
+    HAS_USEDFORSECURITY = False
+
+
+def md5(data: Any, used_for_security: bool | None = None):
+    """Safely allows calling the hashlib.md5 function with the "usedforsecurity" param.
+
+    Args:
+        data (Any): The data to hash.
+        used_for_security (bool, optional): The value to pass to the md5 function's "usedforsecurity" param.
+            Defaults to None.
+
+    Returns:
+        _Hash: The hashed value.
```

Comment:
We use reStructuredText / reST docstrings format ([PEP-287](https://peps.python.org/pep-0287/)) which is used by [Sphinx](http://sphinx-doc.org/) to generate documentation.

Reply:
Thanks for that. Done 👍